vpcs_test.go

```diff
@@ -320,3 +320,121 @@ func TestVPCs_Delete(t *testing.T) {
+func TestVPCs_ListMembers(t *testing.T) {
+	tests := []struct {
+		desc          string
+		expectedQuery string
+		query         interface{}
+		resp          string
+		want          []*VPCMember
+	}{
+		{
+			desc:          "list all members",
+			expectedQuery: "",
+			query:         nil,
+			resp: `{
+		"members": [
+			{
+				"urn": "do:loadbalancer:fb294d78-d193-4cb2-8737-ea620993591b",
+				"name": "nyc1-load-balancer-01",
+				"created_at": "2020-03-16T19:30:48Z"
+			},
+			{
+				"urn": "do:dbaas:13f7a2f6-43df-4c4a-8129-8733267ddeea",
+				"name": "db-postgresql-nyc1-55986",
+				"created_at": "2020-03-15T19:30:48Z"
+			},
+			{
+				"urn": "do:kubernetes:da39d893-96e1-4e4d-971d-1fdda33a46b1",
+				"name": "k8s-nyc1-1584127772221",
+				"created_at": "2020-03-14T19:30:48Z"
+			},
+			{
+				"urn": "do:droplet:86e29982-03a7-4946-8a07-a0114dff8754",
+				"name": "ubuntu-s-1vcpu-1gb-nyc1-01",
+				"created_at": "2020-03-13T19:30:48Z"
+			}
+		],
+		"links": {
+		},
+		"meta": {
+			"total": 4
+		}
+		}`,
+			want: []*VPCMember{
+				{
+					URN:       "do:loadbalancer:fb294d78-d193-4cb2-8737-ea620993591b",
+					Name:      "nyc1-load-balancer-01",
+					CreatedAt: time.Date(2020, 3, 16, 19, 30, 48, 0, time.UTC),
+				},
+				{
+					URN:       "do:dbaas:13f7a2f6-43df-4c4a-8129-8733267ddeea",
+					Name:      "db-postgresql-nyc1-55986",
+					CreatedAt: time.Date(2020, 3, 15, 19, 30, 48, 0, time.UTC),
+				},
+				{
+					URN:       "do:kubernetes:da39d893-96e1-4e4d-971d-1fdda33a46b1",
+					Name:      "k8s-nyc1-1584127772221",
+					CreatedAt: time.Date(2020, 3, 14, 19, 30, 48, 0, time.UTC),
+				},
+				{
+					URN:       "do:droplet:86e29982-03a7-4946-8a07-a0114dff8754",
+					Name:      "ubuntu-s-1vcpu-1gb-nyc1-01",
+					CreatedAt: time.Date(2020, 3, 13, 19, 30, 48, 0, time.UTC),
+				},
+			},
+		},
+		{
+			desc:          "list droplet members",
+			expectedQuery: "droplet",
+			query:         &VPCListMembersRequest{ResourceType: "droplet"},
+			resp: `{
+		"members": [
+			{
+				"urn": "do:droplet:86e29982-03a7-4946-8a07-a0114dff8754",
+				"name": "ubuntu-s-1vcpu-1gb-nyc1-01",
+				"created_at": "2020-03-13T19:30:48Z"
+			}
+		],
+		"links": {
+		},
+		"meta": {
+			"total": 1
+		}
+		}`,
+			want: []*VPCMember{
+				{
+					URN:       "do:droplet:86e29982-03a7-4946-8a07-a0114dff8754",
+					Name:      "ubuntu-s-1vcpu-1gb-nyc1-01",
+					CreatedAt: time.Date(2020, 3, 13, 19, 30, 48, 0, time.UTC),
+				},
+			},
+		},
+	}
+
+	id := "880b7f98-f062-404d-b33c-458d545696f6"
+	path := "/v2/vpcs/" + id + "/members"
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			setup()
+			defer teardown()
+
+			mux.HandleFunc(path, func(w http.ResponseWriter, r *http.Request) {
+				testMethod(t, r, http.MethodGet)
+				require.Equal(t, tt.expectedQuery, r.URL.Query().Get("resource_type"))
+				fmt.Fprint(w, tt.resp)
+			})
+
+			var req *VPCListMembersRequest
+			if tt.query != nil {
+				req = tt.query.(*VPCListMembersRequest)
```

Comment:
Golang question: Why are we doing a type assertion here?

Reply:
Good eye. That is a bit off. In the struct for the testing table, I had specified:

 https://github.com/digitalocean/godo/blob/c533cc33ec04565a4c3a02b25f19659d37a20599/vpcs_test.go#L328

So I needed to do the type assertion to pass it as parameter to the function. Probably simpler to just use `*VPCListMembersRequest` for the type. Pushed an update.